deepdoc/parser/pdf_parser.py

```diff
@@ -956,8 +956,12 @@ def __images__(self, fnm, zoomin=3, page_from=0,
                                 enumerate(self.pdf.pages[page_from:page_to])]
             self.page_images_x2 = [p.to_image(resolution=72 * zoomin * 2).annotated for i, p in
                                 enumerate(self.pdf.pages[page_from:page_to])]
-            self.page_chars = [[{**c, 'top': c['top'], 'bottom': c['bottom']} for c in page.dedupe_chars().chars if self._has_color(c)] for page in
-                               self.pdf.pages[page_from:page_to]]
+            try:
```

Comment:
I did not get it. Why try is in another try?

Reply:
The second try exists to catch and handle exceptions that may occur during parsing of page_chars, so that parsing of the entire PDF does not fail due to parsing problems on individual pages. In the original code, if an exception occurs in page_chars, it is not handled but simply reported as an error and exited.

Reply:
Appreciation!